NitroxTest/Model/Packets/PacketsSerializableTest.cs

```diff
@@ -39,5 +43,59 @@ public void AllPacketsAreSerializable()
                 .ToList()
                 .ForEach(IsSerializable);
         }
+
+        [TestMethod]
+        public void PacketSerializationTest()
+        {
+            SurrogateSelector surrogateSelector = new SurrogateSelector();
+            StreamingContext streamingContext = new StreamingContext(StreamingContextStates.All); // Our surrogates can be safely used in every context.
+
+            Type[] types = typeof(Vector3Surrogate).Assembly
+                .GetTypes();
+
+            IEnumerable<Type> surrogates = types.Where(t =>
+                                                       t.BaseType != null &&
+                                                       t.BaseType.IsGenericType &&
+                                                       t.BaseType.GetGenericTypeDefinition() == typeof(SerializationSurrogate<>) &&
+                                                       t.IsClass &&
+                                                       !t.IsAbstract);
+            foreach (Type type in surrogates)
+            {
+                ISerializationSurrogate surrogate = (ISerializationSurrogate)Activator.CreateInstance(type);
+                Type surrogatedType = type.BaseType.GetGenericArguments()[0];
+                surrogateSelector.AddSurrogate(surrogatedType, streamingContext, surrogate);
+            }
+
+            // For completeness, we could pass a StreamingContextStates.CrossComputer.
+            BinaryFormatter serializer = new BinaryFormatter(surrogateSelector, streamingContext);
```

Comment:
In a follow-up, we may want to pull this out into a factory class as it is a duplicate of the code in `Packet`